src/sendbuf.h

```diff
@@ -22,9 +22,16 @@
 #ifndef SENDBUF_H
 #define SENDBUF_H
 
+typedef void (*ExpandMacro_t) (void);
+
+void ExpandMacro_CurrentQso(void);
+void ExpandMacro_PreviousQso(void);
+
 char short_number(char c);
 void sendmessage(const char *msg);
+void sendmessage_with_macro_expand(const char *msg, ExpandMacro_t expandMacro);
```

Comment:
I think `sendxxx` could expose less of its inner workings. The user just wants to send a message using the current QSO or the previous one. Whether this is implemented using a function pointer or somehow else is internal to `sendbuf`.

Reply:
I'm not sure I understand what specific changes you want to see in this code. Do you mean introducing a function called:
sendmessage_previous_qso and wrap sendmessage_with_macro_expand in it?

Reply:
Yes, practically.
That would also save us from mocking `ExpandMacro_xxx` in test sources not actually related to message sending.